studio/components/interfaces/Organization/Usage/UsageSection.tsx

```diff
@@ -0,0 +1,293 @@
+import * as Tooltip from '@radix-ui/react-tooltip'
+import clsx from 'clsx'
+import Link from 'next/link'
+
+import { USAGE_APPROACHING_THRESHOLD } from 'components/interfaces/BillingV2/Billing.constants'
+import Panel from 'components/ui/Panel'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import SparkBar from 'components/ui/SparkBar'
+import { DataPoint } from 'data/analytics/constants'
+import { ProjectSubscriptionResponse } from 'data/subscriptions/project-subscription-v2-query'
+import {
+  ProjectUsageResponse,
+  UsageMetric,
+  useProjectUsageQuery,
+} from 'data/usage/project-usage-query'
+import { formatBytes } from 'lib/helpers'
+import { Button, IconAlertTriangle, IconBarChart2 } from 'ui'
+import SectionContent from './SectionContent'
+import SectionHeader from './SectionHeader'
+import { CategoryMetaKey, USAGE_CATEGORIES } from './Usage.constants'
+import {
+  ChartTooltipValueFormatter,
+  ChartYFormatterCompactNumber,
+  getUpgradeUrl,
+} from './Usage.utils'
+import UsageBarChart from './UsageBarChart'
+import { ScaffoldContainer, ScaffoldDivider } from 'components/layouts/Scaffold'
+
+interface UsageSectionProps {
+  projectRef: string
+  categoryKey: CategoryMetaKey
+  subscription: ProjectSubscriptionResponse | undefined
+  chartMeta: {
+    [key: string]: { data: DataPoint[]; margin: number; isLoading: boolean; hasNoData: boolean }
+  }
+  currentBillingCycleSelected: boolean
+}
+
+// [Joshen TODO] Needs to change to use org slug, and eventually call useOrganizationUsageQuery
+const UsageSection = ({
+  projectRef,
+  categoryKey,
+  chartMeta,
+  subscription,
+  currentBillingCycleSelected,
+}: UsageSectionProps) => {
+  const { data: usage } = useProjectUsageQuery({ projectRef })
+  const categoryMeta = USAGE_CATEGORIES.find((category) => category.key === categoryKey)
+
+  if (!categoryMeta) return null
+
+  const usageBasedBilling = subscription?.usage_billing_enabled
+  const exceededLimitStyle = !usageBasedBilling ? 'text-red-900' : 'text-amber-900'
+
+  const upgradeUrl = getUpgradeUrl(projectRef, subscription)
+
+  return (
+    <>
+      <ScaffoldContainer>
+        <SectionHeader title={categoryMeta.name} description={categoryMeta.description} />
+      </ScaffoldContainer>
+
+      <ScaffoldDivider />
+
+      {categoryMeta.attributes.map((attribute) => {
+        const usageMeta = usage?.[attribute.key as keyof ProjectUsageResponse] as UsageMetric
+        const usageRatio =
+          typeof usageMeta !== 'number' ? (usageMeta?.usage ?? 0) / (usageMeta?.limit ?? 0) : 0
+        const usageExcess = (usageMeta?.usage ?? 0) - (usageMeta?.limit ?? 0)
+
+        const chartData = chartMeta[attribute.key]?.data ?? []
+
+        const notAllValuesZero =
+          attribute.attributes
+            ?.map((attr) => {
+              return chartData.some((dataPoint) => Number(dataPoint[attr.key]) !== 0)
+            })
+            .some((x) => !!x) ?? false
```

Comment:
How expensive is this? Might be worth moving this into a component so we can useMemo it.

Thinking about it, it might be a good rule of thumb to not do any calculations inside a `.map()` callback function and make a subcomponent (even if it's in the same file). It'll make refactoring easier for us and React can optimize things easier too :)

Reply:
can do - i'll do this in the part 2 PR to prevent conflicts